### ISSUE TITLE
Zero values for MIG instances using dcgm-exporter. 

### ISSUE THREAD

**Issue by Shadowphax** [NVIDIA/dcgm-exporter#3]
Hi, 

**DCGM Version: 2.2.9** 
**CUDA: 11.4**
**Driver: datacenter-gpu-manager-2.2.9-1.x86_64**

We have recently purchased a Dell R750xa with x4 A100-40GB GPUs. I built the dcgm-exporter binary from source and when running can obtain values of the parent GPU cards. However, all values are set as zero for MIG instances even though we have utilization on the MIG instances. 

I have also noticed that not all the MIG profiles are listed.  

Thank You !!

**Comment by nikkon-dev** [comment]
Hi @Shadowphax,

Have you tried to run dcgm-exporter with `-f` command line argument?

**Comment by Shadowphax** [comment]
Hi @nikkon-dev 

Appreciate the feedback, thank you. 

dcgm-exporter -f /etc/dcgm-exporter/default-counters.csv --devices g:2,3,i:0
INFO[0000] Starting dcgm-exporter                       
FATA[0000] Invalid ranged device option 'g:2,3,i:0': there can only be one specified range

I am not sure if I have the syntax correct? Also, MIG IDs (instance IDs) are required as input for **"i"**. Are we meant to use the ID from **"nvidia-smi"** for the instance ID?

**Comment by nikkon-dev** [comment]
@Shadowphax,
You are right. In the current implementation, dcgm-exporter does not support multiple types of ranges and more than one range in general. This limitation is on the dcgm-exporter side only, as DCGM itself supports such scenarios.

**Comment by Shadowphax** [comment]
Hi @nikkon-dev 

Could you provide guidance on how to monitor a single MIG instance with **dcgm-exporter**, especially to identify the correct MIG instance IDs?  I've partitioned the cards appropriately. Is there something I am meant to do with **dcgmi** in order for **dcgm-exporter** to expose the metrics for MIG? 

**dcgmi group -l** 
+-------------------+----------------------------------------------------------+
| GROUPS                                                                       |
| 2 groups found.                                                              |
+===================+==========================================================+
| Groups            |                                                          |
| -> 2              |                                                          |
|    -> Group ID    | 2                                                        |
|    -> Group Name  | card0                                                    |
|    -> Entities    | GPU 0, GPU_I 3, GPU_I 2, GPU_I 0, GPU_I 1                |
| -> 3              |                                                          |
|    -> Group ID    | 3                                                        |
|    -> Group Name  | card1                                                    |
|    -> Entities    | GPU 1, GPU_I 10, GPU_I 7, GPU_I 8, GPU_I 9               |
+-------------------+----------------------------------------------------------+

So I have four MIG instances on each card. I would like to know where the ID is obtained per MIG for **--devices i:(x)** where x is either 0 or 1 as mentioned in --help 

Cheers

**Comment by xwhuang0923** [comment]
@nikkon-dev 
Hi, I got zero value for MIG instance too.
CUDA: 11.4
Driver: datacenter-gpu-manager  1:2.3.1   amd64
![1637292603(1)](https://user-images.githubusercontent.com/20106562/142560168-a0a95bac-2c32-4f97-89d4-0ada3a0b048a.png)

dcgm-exporter --devices=i:2

![1637292721(1)](https://user-images.githubusercontent.com/20106562/142560362-57dabf8b-0c2b-4d8d-a27d-41a2d933b61b.png)


![1637292787(1)](https://user-images.githubusercontent.com/20106562/142560453-26103d98-e99c-4cfc-b307-a7205861549b.png)

Thank you!

**Comment by nikkon-dev** [comment]
@xwhuang0923,

Please take a look at the `dcgmi discovery -c` output. In the `--device=i:X` argument, the `X` is the entity ID from the discovery command output, not the MIG Dev from the `nvidia-smi` output.

WBR,
Nik

**Comment by xwhuang0923** [comment]
@nikkon-dev 
Here is the output:

![image](https://user-images.githubusercontent.com/20106562/142718780-6ab8f5f9-7d6e-469a-ad71-9144887d71f4.png)

**Comment by glowkey** [comment]
We have been making quite a few enhancements to DCGM and DCGM-Exporter with respect to MIG. Some of these fixes/enhancements are available in 2.3.2-2.6.3 and others are coming in next version as well. Please try upgrading now or in a few weeks and letting us know if these latest versions do not fix the issue.

**Comment by STALLAAAA** [comment]
@glowkey Is there any example for mig instance states monitoring such as insatance memory or instance utilization?

**Comment by nccurry** [comment]
Closing as stale. Please reopen against the latest version if this is still an issue.
